### PR TITLE
Assorted Makefile changes

### PR DESCRIPTION
**Pull request by sungeunchoi**
- Remove qthreads include path from RUNTIME_CFLAGS (redundant).  This
  can probably be done for massivethreads also.

- Fix serialization of make -j in the examples directory.  It wasn't
  actually too bad, since the directory structure is not too deep, but
  the warning messages were annoying.

- Add a Makefile in the patterns directory and add it to the top-level
  Makefile.

- Remove ldconfig check for fftw.  I'm not sure what the original
  motivation for this was, but it wasn't a robust solution (and also
  was broken on XC).

@mppf @bradcray 

Passes relevant testing: runtime build for qthreads, make -j without warnings, make on XC with and without fftw.

### CODE REVIEW

**Comment by daviditen** [comment]
What are the serialization warnings for 'make -j'?  I don't see warnings about that with "make -j" in examples/ on my computer.

**Comment by sungeunchoi** [comment]
That's strange.  I believe others also got this error.
```
warning: jobserver unavailable: using -j1. Add `+' to parent make rule.
```

File: test/release/examples/Makefile [hunk 1]
```diff
@@ -26,13 +26,16 @@ clean: FORCE
 benchmarks: FORCE
 	cd benchmarks && $(MAKE)
 
+patterns: FORCE
```
Comment (daviditen) [inline test/release/examples/Makefile]: Could you please add the "patterns" directory to the "clean:" target in this Makefile?

Comment (daviditen) [inline test/release/examples/Makefile]: I switched to a different system and see the "jobserver unavailable" warnings now - and also that this PR fixes them.

I think this is good to go. Thanks.

**Pull request merged**